Lib/test/test_ntpath.py

```diff
@@ -117,6 +117,35 @@ def test_splitdrive(self):
         # Issue #19911: UNC part containing U+0130
         self.assertEqual(ntpath.splitdrive('//conky/MOUNTPOİNT/foo/bar'),
                          ('//conky/MOUNTPOİNT', '/foo/bar'))
+        # gh-81790: support device namespace, including UNC drives.
+        tester('ntpath.splitdrive("//?/c:")', ("//?/c:", ""))
+        tester('ntpath.splitdrive("//?/c:/")', ("//?/c:", "/"))
+        tester('ntpath.splitdrive("//?/c:/dir")', ("//?/c:", "/dir"))
+        tester('ntpath.splitdrive("//?/UNC")', ("", "//?/UNC"))
+        tester('ntpath.splitdrive("//?/UNC/")', ("", "//?/UNC/"))
+        tester('ntpath.splitdrive("//?/UNC/server/")', ("//?/UNC/server/", ""))
+        tester('ntpath.splitdrive("//?/UNC/server/share")', ("//?/UNC/server/share", ""))
+        tester('ntpath.splitdrive("//?/UNC/server/share/dir")', ("//?/UNC/server/share", "/dir"))
+        tester('ntpath.splitdrive("//?/VOLUME{00000000-0000-0000-0000-000000000000}/spam")',
+               ('//?/VOLUME{00000000-0000-0000-0000-000000000000}', '/spam'))
+        tester('ntpath.splitdrive("//?/BootPartition/")', ("//?/BootPartition", "/"))
+        tester('ntpath.splitdrive("//?/Harddisk0Partition2/")', ("//?/Harddisk0Partition2", "/"))
+        tester('ntpath.splitdrive("//?/HarddiskVolume2/")', ("//?/HarddiskVolume2", "/"))
```

Comment:
I think these two cases are unnecessary. The implementation special cases the "UNC" device. The behavior for other device names is tested by the "Volume{GUID}" and "BootPartition" cases.

Reply:
I know it doesn't hurt technically in terms of correctness, but extra tests that aren't needed slow the test suite down for no reason. I can't imagine a change that would break these cases and not also break the "BootPartition" case. I see no reason for the implementation of `splitdrive()` to special case undocumented volume names such as "BootPartition" and "HarddiskVolume2", as opposed to documented drive-letter and "Volume{GUID}" volume names.